golang/cosmos/ante/inbound.go

```diff
@@ -0,0 +1,73 @@
+package ante
+
+import (
+	"github.com/armon/go-metrics"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	swingtypes "github.com/Agoric/agoric-sdk/golang/cosmos/x/swingset/types"
+)
+
+// TODO: We don't have a more appropriate error type for this.
+var ErrInboundQueueFull = sdkerrors.ErrMempoolIsFull
+
+type inboundAnte struct {
+	sk SwingsetKeeper
+}
+
+// NewInboundDecorator return an AnteDecorator which honors the allowed size of the inbound queue.
+// The swingset message types will consume one allowed entry each, and will be reflected
+// in the action queue when executed.
+func NewInboundDecorator(sk SwingsetKeeper) sdk.AnteDecorator {
+	return inboundAnte{sk: sk}
+}
+
+// AnteHandle implements sdk.AnteDecorator
+func (ia inboundAnte) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	msgs := tx.GetMsgs()
+	inboundsAllowed := int32(-1)
+	for _, msg := range msgs {
+		switch msg.(type) {
+		case *swingtypes.MsgDeliverInbound,
+			*swingtypes.MsgInstallBundle,
+			*swingtypes.MsgProvision,
+			*swingtypes.MsgWalletAction,
+			*swingtypes.MsgWalletSpendAction:
+			if inboundsAllowed == -1 {
+				state := ia.sk.GetState(ctx)
+				allowed, found := swingtypes.QueueSizeEntry(state.QueueAllowed, "inbound")
```

Comment:
Does `allowed` mean the maximum (1000), or the remaining space? The `allowed - actions` below suggests that it means the maximum, but I'm not sure that's how it was used over in the JS side.